pkg/objstore/timeout.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package objstore
+
+import (
+	"context"
+	"io"
+	"time"
+)
+
+var DefaultTimeout = Timeout{
+	Iter:       5 * time.Minute,
+	Get:        5 * time.Second,
+	GetRange:   30 * time.Second,
+	Exists:     5 * time.Second,
+	Upload:     10 * time.Minute,
+	Delete:     5 * time.Second,
+	Attributes: 1 * time.Minute,
+}
+
+type Timeout struct {
+	Iter       time.Duration `yaml:"iter"`
```

Comment:
I'm looking for feedbacks for the default timeout values.

Reply:
maybe default disabled?

Reply:
Yes, definitely could be a good strategy.

Reply:
> As I mentioned in one of the previous Thanos community calls, I personally believe that whenever there's a data transfer involved (download/upload) we should timeout based on minimum speed and not time. It's very difficult to fine-tune these timeouts otherwise.

This is also on my agenda. I wanted to have a simpler solution as well. But you have a fair point that it'd be harder to come up with those default values. I'll park this one and focus on bandwidth/minimum speed based implementation and then we can decide which one to go with.

Reply:
ooh yeah I very much like this idea as well!